### PR TITLE
HDKeys: add toBuffer/fromBuffer

### LINKED ISSUE #741: HDKeys Standard Methods G7
HDPublicKey:
- [ ] `fromBuffer()`
- [x] `fromString()`
- [x] `fromJSON()`
- [ ] `toBuffer()`

HDPrivateKey:
- [ ] `fromBuffer()`
- [x] `fromString()`
- [x] `fromJSON()`
- [ ] `toBuffer()`

### PR DESCRIPTION
**Pull request by eordano**
Closes #741

### CODE REVIEW

**Comment by braydonf** [comment]
LGTM

**Comment by eordano** [comment]
Added missing jsdocs

File: lib/hdprivatekey.js [hunk 1]
```diff
@@ -480,10 +480,34 @@ HDPrivateKey.prototype.toObject = function toObject() {
...
+  return new HDPrivateKey(arg);
+};
+
+/**
+ * Returns a buffer representation of the HDPrivateKey
+ *
+ * @return {string}
+ */
+HDPrivateKey.prototype.toBuffer = function() {
+  return this._buffers.xprivkey;
```
Comment (maraoz) [inline lib/hdprivatekey.js]: shouldn't this return a copy `Buffer` instead of a reference?

Comment (eordano) [inline lib/hdprivatekey.js]: Addressed

**Pull request merged**